t/js_saml.t

```diff
@@ -0,0 +1,1642 @@
+        </saml:Conditions>
+        <saml:AuthnStatement AuthnInstant=""
+                            SessionNotOnOrAfter=""
+                            SessionIndex="_nginx_sessionindex_"
+                            >
+            <saml:AuthnContext>
+                <saml:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:Password</saml:AuthnContextClassRef>
+            </saml:AuthnContext>
+        </saml:AuthnStatement>
+        <saml:AttributeStatement>
+            <saml:Attribute Name="uid"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">1</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="memberOf"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">group1, admins, students</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="email"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">user1</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="name"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">Alan Alda</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="telephoneNumber"
+                            NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">+31(0)12345678</saml:AttributeValue>
+            </saml:Attribute>
+            <saml:Attribute Name="http://schemas.example.com/identity/claims/foo"
+                            >
+                <saml:AttributeValue xsi:type="xs:string">bar</saml:AttributeValue>
+            </saml:Attribute>
+        </saml:AttributeStatement>
+    </saml:Assertion>
+</samlp:Response>
+END_XML
+
+    my $logout_request = <<"END_XML";
+<samlp:LogoutRequest xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+                     xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
+                     ID=""
+                     Version="2.0"
+                     IssueInstant=""
+                     Destination=""
+                     NotOnOrAfter=""
+                     >
+    <saml:Issuer></saml:Issuer>
+    $signature
+    <saml:NameID SPNameQualifier=""
+                 Format="urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
+                 >user1</saml:NameID>
+    <samlp:SessionIndex>_nginx_sessionindex</samlp:SessionIndex>
+</samlp:LogoutRequest>
+END_XML
+
+my $logout_response = <<"END_XML";
```

Comment:
You don't have to wrap END_XML into quotes.

Reply:
True, fixed!